### PR TITLE
Correct 'XCTAssertEqual(_:_:)' argument order.

### LINKED ISSUE #153: Correct 'XCTAssertEqual(_:_:)' argument order.
#109 moved all TensorFlow APIs from apple/swift to this repository. However, some test assertions weren't updated because the argument order of the `expectEqual(_:_:)` function is different from that of `XCTAssertEqual(_:_:)` in that `expectEqual(_:_:)` expected the **first** argument to be the expected value, while `XCTAssertEqual(_:_:)` expected the **second** argument.

Here's a concrete example of a test assertion that used the old order:
https://github.com/tensorflow/swift-apis/blob/f194110443638f9f1a989307f21e43b653a5deb0/Tests/TensorFlowTests/OperatorTests/MathTests.swift#L22

It should be changed to:
```swift
XCTAssertEqual(x.sum(), Tensor(30))
```

### Task
Go through all tests (especially the ones in [`OperatorTests/`](https://github.com/tensorflow/swift-apis/tree/master/Tests/TensorFlowTests/OperatorTests)) and make sure `XCTAssertEqual(_:_:)` arguments are ordered correctly: The first argument should be the value to be tested, and the second argument should be the expected value.

### PR DESCRIPTION
**Pull request by Shashi456**
fix #153

### CODE REVIEW

**Comment by Shashi456** [comment]
Please do tell me, If I have missed anything. I'll request a review once i'm done with all the files in the directory

**Comment by Shashi456** [comment]
@rxwei @eaplatanios requesting a review. Most probably only cleanup needed,

**Comment by Shashi456** [comment]
Also, I did skim the rest of the tests, there seem to be no errors in them.

File: Tests/TensorFlowTests/OperatorTests/BasicTests.swift [hunk 1]
```diff
@@ -287,26 +288,27 @@ final class BasicOperatorTests: XCTestCase {
...
         /// Test shapes
-        XCTAssertEqual([1, 4, 5], array3D.shape)
-        XCTAssertEqual([2, 5], array2D.shape)
-        XCTAssertEqual([2], array1D.shape)
+        XCTAssertEqual(array3D.shape, [1, 4, 5])
+        XCTAssertEqual(array2D.shape, [2, 5])
+        XCTAssertEqual(array1D.shape, [2])
 
         /// Test scalars
         XCTAssertEqual(
```
Comment (Shashi456) [inline Tests/TensorFlowTests/OperatorTests/BasicTests.swift]: @rxwei @eaplatanios this seems to be throwing the build error, should I wrap the addition in a parentheses of its own?

Comment (Shashi456) [inline Tests/TensorFlowTests/OperatorTests/BasicTests.swift]: @rxwei @eaplatanios this seems to be throwing the build error, should I wrap the addition in a parentheses of its own?

Comment (rxwei) [reply Tests/TensorFlowTests/OperatorTests/BasicTests.swift]: You can define the expected value as a separate `let` variable.

Comment (eaplatanios) [reply Tests/TensorFlowTests/OperatorTests/BasicTests.swift]: I think that changing `Array` to `Array<Int>` may help the compiler resolve this faster.

Comment (Shashi456) [reply Tests/TensorFlowTests/OperatorTests/BasicTests.swift]: Oh no, I figured out the error, I wrote a `shape` instead of `scalars`.

**Comment by Shashi456** [comment]
@rxwei There's just so many PRs xD. Can you trigger a build on this?

**Comment by Shashi456** [comment]
Alright, all the builds seem to be failing because of the error on line 296. This should fix that.

**Comment by Shashi456** [comment]
@rxwei so the error is that `expected` is not double, any way i could get it to conform to that?

**Comment by rxwei** [comment]
Apply the following change will fix the error.

```diff
-               let expected = Array(stride(from: 20.0, to: 30, by: 2)) +
-                              Array(stride(from: 45.0, to: 50, by: 1)) +
-                       Array(stride(from: 30.0, to: 40, by: 2)) +
-                       Array(stride(from: 55.0, to: 60, by: 1))
-
         /// Test shapes
         XCTAssertEqual(array3D.shape, [1, 4, 5])
         XCTAssertEqual(array2D.shape, [2, 5])
         XCTAssertEqual(array1D.shape, [2])

         /// Test scalars
-        XCTAssertEqual(array3D.scalars, expected)
+        XCTAssertEqual(array3D.scalars,
+                       [Float](stride(from: 20.0, to: 30, by: 2)) +
+                            Array(stride(from: 45.0, to: 50, by: 1)) +
+                       [Float](stride(from: 30.0, to: 40, by: 2)) +
+                       [Float](stride(from: 55.0, to: 60, by: 1)))
         XCTAssertEqual(array2D.scalars, Array(stride(from: 20.0, to: 30, by: 1)))
         XCTAssertEqual(array1D.scalars, Array(stride(from: 3.0, to: 5, by: 1)))
     }
```

**Comment by Shashi456** [comment]
@rxwei  @eaplatanios Can you help me debug this ?
```
[86/86] Testing TensorFlowTests.MathOperatorTests/testBroadcastedAddGradient

Test Suite 'Selected tests' started at 2019-05-31 14:27:03.685
Test Suite 'SequentialTests' started at 2019-05-31 14:27:04.877
Test Case 'SequentialTests.testSequential' started at 2019-05-31 14:27:04.877
Exited with signal code 11

Test Suite 'Selected tests' started at 2019-05-31 14:27:03.636
Test Suite 'TrivialModelTests' started at 2019-05-31 14:27:04.716
Test Case 'TrivialModelTests.testXOR' started at 2019-05-31 14:27:04.716
Exited with signal code 11

Test Suite 'Selected tests' started at 2019-05-31 14:27:17.778
Test Suite 'LayerTests' started at 2019-05-31 14:27:18.919
Test Case 'LayerTests.testRNN' started at 2019-05-31 14:27:18.919
Exited with signal code 11

Test Suite 'Selected tests' started at 2019-05-31 14:27:33.903
Test Suite 'DatasetTests' started at 2019-05-31 14:27:34.889
Test Case 'DatasetTests.testSingleValueTransformations' started at 2019-05-31 14:27:34.889
failed to demangle witness for associated type 'Iterator' in conformance 'TensorFlow.Dataset<TensorFlow.Tensor<Swift.Float>>: Sequence' from mangled name '10TensorFlow15DatasetIteratorVyxG'
Exited with signal code 6

Test Suite 'Selected tests' started at 2019-05-31 14:27:34.047
Test Suite 'DatasetTests' started at 2019-05-31 14:27:35.155
Test Case 'DatasetTests.testSingleValueHOFs' started at 2019-05-31 14:27:35.155
failed to demangle witness for associated type 'Iterator' in conformance 'TensorFlow.Dataset<TensorFlow.Tensor<Swift.Float>>: Sequence' from mangled name '10TensorFlow15DatasetIteratorVyxG'
Exited with signal code 6

Test Suite 'Selected tests' started at 2019-05-31 14:27:34.697
Test Suite 'DatasetTests' started at 2019-05-31 14:27:35.849
Test Case 'DatasetTests.testParallelMap' started at 2019-05-31 14:27:35.849
failed to demangle witness for associated type 'Iterator' in conformance 'TensorFlow.Dataset<TensorFlow.Tensor<Swift.Float>>: Sequence' from mangled name '10TensorFlow15DatasetIteratorVyxG'
Exited with signal code 6

Test Suite 'Selected tests' started at 2019-05-31 14:27:35.076
Test Suite 'DatasetTests' started at 2019-05-31 14:27:36.272
Test Case 'DatasetTests.testMapToDifferentType' started at 2019-05-31 14:27:36.272
failed to demangle witness for associated type 'Iterator' in conformance 'TensorFlow.Dataset<TensorFlow.Tensor<Swift.Float>>: Sequence' from mangled name '10TensorFlow15DatasetIteratorVyxG'
Exited with signal code 6

Test Suite 'Selected tests' started at 2019-05-31 14:27:36.426
Test Suite 'MathOperatorTests' started at 2019-05-31 14:27:37.526
Test Case 'MathOperatorTests.testExpm1' started at 2019-05-31 14:27:37.527
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("1.7182817") is not equal to ("1.0") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("6.389056") is not equal to ("2.7182817") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("19.085537") is not equal to ("7.389056") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("53.598152") is not equal to ("20.085537") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("147.41316") is not equal to ("54.598152") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("1.7182817") is not equal to ("1.0") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("6.389056") is not equal to ("2.7182817") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("19.085537") is not equal to ("7.389056") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("53.598152") is not equal to ("20.085537") +/- ("0.0001") -
/home/paws/swift-apis/Tests/TensorFlowTests/Helpers.swift:20: error: MathOperatorTests.testExpm1 : XCTAssertEqual failed: ("147.41316") is not equal to ("54.59815") +/- ("0.0001") -
Test Case 'MathOperatorTests.testExpm1' failed (0.136 seconds)
Test Suite 'MathOperatorTests' failed at 2019-05-31 14:27:37.662
         Executed 1 test, with 10 failures (0 unexpected) in 0.136 (0.136) seconds
Test Suite 'Selected tests' failed at 2019-05-31 14:27:37.663
         Executed 1 test, with 10 failures (0 unexpected) in 0.136 (0.136) seconds
```

**Comment by eaplatanios** [comment]
This is a bug that appears in Linux and not Mac. You can track its progress in #161.

**Comment by Shashi456** [comment]
and the math errors @eaplatanios, can you also check if the values are right or wrong?

**Comment by Shashi456** [comment]
the dataset errors persist. I thought it would be because of the order. but the reason seems to be something else.

**Comment by eaplatanios** [comment]
@Shashi456 The numerical errors in the tests should be fixed in #147. I'm not sure why we get the other failures on Linux yet.

**Comment by Shashi456** [comment]
@eaplatanios are you saying the dataset tests fail only on linux?

**Comment by eaplatanios** [comment]
I believe so yes. At least they don't fail on my MacBook.

**Comment by Shashi456** [comment]
@eaplatanios, that's very strange

File: Tests/TensorFlowTests/OperatorTests/DatasetTests.swift [hunk 2]
```diff
@@ -91,18 +91,18 @@ final class DatasetTests: XCTestCase {
...
-        XCTAssertEqual([0, 2, 4], evens.flatMap { $0.scalars })
+        XCTAssertEqual([0, 2, 4], evens.flatMap{ $0.scalars })
     }
 
     func testParallelMap() {
         let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
         let dataset = Dataset(elements: scalars)
         let addedOne: Dataset = dataset.map(parallelCallCount: 5) { $0 + 1 }
-        XCTAssertEqual([1, 2, 3, 4, 5], addedOne.flatMap { $0.scalars })
+        XCTAssertEqual([1, 2, 3, 4, 5], addedOne.flatMap{ $0.scalars })
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]:
```suggestion
        XCTAssertEqual(addedOne.flatMap { $0.scalars }, [1, 2, 3, 4, 5])
```

File: Tests/TensorFlowTests/OperatorTests/DatasetTests.swift [hunk 3]
```diff
@@ -80,7 +80,7 @@ final class DatasetTests: XCTestCase {
         let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
         let dataset = Dataset(elements: scalars)
         let shuffled = dataset.shuffled(sampleCount: 5, randomSeed: 42)
-        XCTAssertEqual([0, 4, 1, 3, 2], shuffled.map { $0.scalar! })
+        XCTAssertEqual([0, 4, 1, 3, 2], shuffled.map{ $0.scalar! })
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]: 1. The horizontal spacing was already correct. The change made it incorrect. Same for the next few changes.
2. Arguments need to be swapped.

```suggestion
        XCTAssertEqual(shuffled.map { $0.scalar! }, [0, 4, 1, 3, 2])
```

File: Tests/TensorFlowTests/OperatorTests/DatasetTests.swift [hunk 4]
```diff
@@ -91,18 +91,18 @@ final class DatasetTests: XCTestCase {
         // Use '.==' in the following closure to avoid any conversions to
         // host data types, which is not handled correctly in tracing.
         let evens: Dataset = dataset.filter { Tensor($0 % 2) .== Tensor(0) }
-        XCTAssertEqual([0, 2, 4], evens.flatMap { $0.scalars })
+        XCTAssertEqual([0, 2, 4], evens.flatMap{ $0.scalars })
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]:
```suggestion
        XCTAssertEqual(evens.flatMap { $0.scalars }, [0, 2, 4])
```

File: Tests/TensorFlowTests/OperatorTests/DatasetTests.swift [hunk 5]
```diff
@@ -91,18 +91,18 @@ final class DatasetTests: XCTestCase {
...
         let scalars = Tensor<Float>(rangeFrom: 0, to: 5, stride: 1)
         let dataset = Dataset(elements: scalars)
         let addedOne: Dataset = dataset.map(parallelCallCount: 5) { $0 + 1 }
-        XCTAssertEqual([1, 2, 3, 4, 5], addedOne.flatMap { $0.scalars })
+        XCTAssertEqual([1, 2, 3, 4, 5], addedOne.flatMap{ $0.scalars })
         // Use '.==' in the following closure to avoid any conversions to
         // host data types, which is not handled correctly in tracing.
         let evens: Dataset = dataset.filter { Tensor($0 % 2) .== Tensor(0) }
-        XCTAssertEqual([0, 2, 4], evens.flatMap { $0.scalars })
+        XCTAssertEqual([0, 2, 4], evens.flatMap{ $0.scalars })
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]:
```suggestion
        XCTAssertEqual(evens.flatMap { $0.scalars }, [0, 2, 4])
```

File: Tests/TensorFlowTests/OperatorTests/DatasetTests.swift [hunk 6]
```diff
@@ -111,7 +111,7 @@ final class DatasetTests: XCTestCase {
         let shuffled = dataset.shuffled(sampleCount: 5, randomSeed: 42)
         XCTAssertEqual([0, 4, 1, 3, 2], shuffled.map { $0.scalar! })
         let evens = shuffled.map { Tensor($0 % 2) .== Tensor(0) }
-        XCTAssertEqual([true, true, false, false, true], evens.map { $0.scalar! })
+        XCTAssertEqual([true, true, false, false, true], evens.map { $0.scalar! } )
```
Comment (rxwei) [inline Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]:
```suggestion
        XCTAssertEqual(evens.map { $0.scalar! }, [true, true, false, false, true])
```

Comment (Shashi456) [reply Tests/TensorFlowTests/OperatorTests/DatasetTests.swift]: @rxwei I will make these changes, but when i tried to locally build it before reverting these changes, This didn't pass either.

**Comment by Shashi456** [comment]
@rxwei I tried to test this locally, the errors persist. I'm using nightly.

**Comment by Shashi456** [comment]
So @rxwei just a small question, Do you think this is a issue within the code? because @eaplatanios did mention that they are passing on mac OS. I'm inclined to believe it's an ubuntu issue,

**Comment by Shashi456** [comment]
@rxwei could you trigger a build on this once? I want to see if #163 solved the issues.

**Comment by Shashi456** [comment]
@rxwei this passes now, I think this can be merged. Thanks a lot, @bgogul for fixing the errors.

**Comment by rxwei** [comment]
Thanks!

**Pull request merged**